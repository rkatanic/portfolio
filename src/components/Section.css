.section {
  display: flex;
  flex-direction: column;
  padding: 2rem 0;
  min-height: 100vh;
  max-width: var(--width-desktop);
}
