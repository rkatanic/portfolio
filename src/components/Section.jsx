import "./Section.css";

const Section = (props) => (
  <section id={props.id} className="section">
    {props.children}
  </section>
);

export default Section;
