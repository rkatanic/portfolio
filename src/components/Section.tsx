import "./Section.css";

interface Props {
  id: string;
  children: JSX.Element | JSX.Element[];
}

const Section = ({ id, children }: Props): JSX.Element => (
  <section id={id} className="section">
    {children}
  </section>
);

export default Section;
