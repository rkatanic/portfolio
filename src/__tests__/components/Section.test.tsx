import { render } from "@testing-library/react";
import Section from "../../components/Section";

describe("Section", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(
      <Section id="section-test">
        <div>test page</div>
      </Section>
    );

    expect(baseElement).toMatchSnapshot();
  });
});
