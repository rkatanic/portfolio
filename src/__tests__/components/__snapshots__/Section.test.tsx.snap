// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should render 1`] = `
<body>
  <div>
    <section
      class="section"
      id="section-test"
    >
      <div>
        test page
      </div>
    </section>
  </div>
</body>
`;
